src/main/java/pl/edu/icm/pl/mxrdr/extension/importer/cif/file/internal/CifMetadataReader.java

```diff
@@ -0,0 +1,122 @@
+                .add(MxrdrMetadataField.PDB_DOI, readValues(MmCifBlock::getCitation, Citation::getPdbxDatabaseIdDOI))
+                .add(MxrdrMetadataField.PDB_STRUCTURE_AUTHOR, readValues(MmCifBlock::getAuditAuthor, AuditAuthor::getName))
+                .add(MxrdrMetadataField.PDB_DEPOSIT_DATE, readValues(MmCifBlock::getPdbxDatabaseStatus, PdbxDatabaseStatus::getRecvdInitialDepositionDate))
+                .add(MxrdrMetadataField.PDB_RELEASE_DATE, readValues(MmCifBlock::getPdbxAuditRevisionHistory, PdbxAuditRevisionHistory::getRevisionDate))
+                .add(MxrdrMetadataField.PDB_REVISION_DATE, readValues(MmCifBlock::getPdbxAuditRevisionHistory, PdbxAuditRevisionHistory::getRevisionDate))
+                .add(MxrdrMetadataField.CITATION_TITLE, zipLists(citationOrder, readValues(MmCifBlock::getCitation, Citation::getTitle), CifFileConstants.SAFE_ITEMS_DELIMITER))
+                .add(MxrdrMetadataField.CITATION_PUBMED_ID, zipLists(citationOrder, readValues(MmCifBlock::getCitation, Citation::getPdbxDatabaseIdPubMed), CifFileConstants.SAFE_ITEMS_DELIMITER))
+                .add(MxrdrMetadataField.CITATION_JOURNAL, zipLists(citationOrder, readValues(MmCifBlock::getCitation, Citation::getJournalAbbrev), CifFileConstants.SAFE_ITEMS_DELIMITER))
+                .add(MxrdrMetadataField.CITATION_YEAR, zipLists(citationOrder, readValues(MmCifBlock::getCitation, Citation::getYear), CifFileConstants.SAFE_ITEMS_DELIMITER))
+                .add(MxrdrMetadataField.CITATION_AUTHOR, zipLists(
+                        readValues(MmCifBlock::getCitationAuthor, CitationAuthor::getCitationId),
+                        readValues(MmCifBlock::getCitationAuthor, CitationAuthor::getName), CifFileConstants.SAFE_ITEMS_DELIMITER));
+    }
+
+    <T> List<String> readValues(Function<MmCifBlock, T> categoryGetter,
+                                        Function<T, Column<?>> itemGetter) {
+        return Optional.ofNullable(categoryGetter.apply(mainBlock))
+                .map(itemGetter)
+                .filter(Column::isDefined)
+                .map(Column::stringData)
+                .orElseGet(Stream::empty)
+                .map(v -> "?".equals(v) || ".".equals(v) ? StringUtils.EMPTY : v)
+                .collect(Collectors.toList());
+    }
+
+    List<String> zipLists(List<String> first, List<String> second, String delimiter) {
+        int firstSize = first.size();
+        int secondSize = second.size();
+
+        if (firstSize != secondSize) {
+            List<String> listToExtend = firstSize > secondSize ? second : first;
+            int difference = Math.abs(firstSize - secondSize);
+            for (int i = 0; i < difference; i++) {
+                listToExtend.add(StringUtils.EMPTY);
+            }
+        }
+
+        return IntStream.range(0, first.size()) // need to call again as the list might have been extended
+                .mapToObj(i -> first.get(i) + delimiter + second.get(i))
```

Comment:
Joining values here only to filter them out and split later seems overcomplicated. Why not only get what we need in the first place?